File: Python/1046. Last Stone Weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones.sort()
        while(len(stones) > 1):
            stones.sort()
            x = stones.pop()
            y = stones.pop()
            stones.append(abs(x - y))
        return stones[0]

File: Scala/1046. Last Stone Weight.scala
object Solution {
  def lastStoneWeight(stones: Array[Int]): Int = {
    var stack = scala.collection.mutable.Stack[Int](stones:_*).reverse

    while(stack.size > 1){
      stack = stack.sorted.reverse
      val x = stack.pop()
      val y = stack.pop()
      stack.push((x - y).abs)
    }
      stack(0)
  }
}
